### PR TITLE
HTBHF-955 Add smoke tests.

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: swagger.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ info:
   license:
     name: "MIT"
     url: "https://opensource.org/licenses/MIT"
-host: "localhost:38199"
+host: "localhost:39413"
```
Comment (YetAnotherMatt) [inline swagger.yml]: we should add `.host("N/A")` to ApiDocumentation

Comment (dwybourn) [reply swagger.yml]: Done.

File: smoke_tests/src/test/java/uk/gov/dhsc/htbhf/smoke/ServiceIsRunningSpec.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.net.URI;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.springframework.http.HttpStatus.OK;
+
+@ExtendWith(SpringExtension.class)
+class ServiceIsRunningSpec {
```
Comment (mikeSmith08) [inline smoke_tests/src/test/java/uk/gov/dhsc/htbhf/smoke/ServiceIsRunningSpec.java]: Can we call this `ServiceIsRunningTest` and not spec?

File: smoke_tests/src/test/java/uk/gov/dhsc/htbhf/smoke/ServiceIsRunningSpec.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    private static final String BASE_URL = System.getProperty("base_url") == null ? "http://localhost:8080" : System.getProperty("base_url");
+    private static final String PROTOCOL = BASE_URL.startsWith("http") ? "" : "https://";
+    private static final URI HEALTH_URI = URI.create(PROTOCOL + BASE_URL + "/actuator/health");
+
+    private TestRestTemplate client = new TestRestTemplate();
+
+    @Test
+    void healthEndpointShouldReturnUp() {
+        System.out.println("Endpoint is: " + HEALTH_URI.toString());
```
Comment (mikeSmith08) [inline smoke_tests/src/test/java/uk/gov/dhsc/htbhf/smoke/ServiceIsRunningSpec.java]: Don't want this anymore.

File: smoke_tests/src/test/java/uk/gov/dhsc/htbhf/smoke/ServiceIsRunningSpec.java (hunk 2 above)
Comment (dwybourn) [reply smoke_tests/src/test/java/uk/gov/dhsc/htbhf/smoke/ServiceIsRunningSpec.java]: Renamed

File: smoke_tests/src/test/java/uk/gov/dhsc/htbhf/smoke/ServiceIsRunningSpec.java (hunk 3 above)
Comment (dwybourn) [reply smoke_tests/src/test/java/uk/gov/dhsc/htbhf/smoke/ServiceIsRunningSpec.java]: Removed.

File: swagger.yml (hunk 1 above)
Comment (YetAnotherMatt) [reply swagger.yml]: you need to re-run the build too I'm afraid, so that this changes to N/A

**Review by YetAnotherMatt — DISMISSED** [review]

**Comment by mikeSmith08** [comment]
You still need to fix the `.gitmodules` file by adding in the specific branch (master).

**Review by YetAnotherMatt — DISMISSED** [review]

File: api/src/test/java/uk/gov/dhsc/htbhf/card/testhelper/CardDTOTestDataFactory.java [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 import java.util.UUID;
 
-import static uk.gov.dhsc.htbhf.card.helper.AddressDTOTestDataFactory.aValidAddress;
-import static uk.gov.dhsc.htbhf.card.helper.TestConstants.HOMER_DOB;
-import static uk.gov.dhsc.htbhf.card.helper.TestConstants.HOMER_EMAIL;
-import static uk.gov.dhsc.htbhf.card.helper.TestConstants.HOMER_FIRST_NAME;
-import static uk.gov.dhsc.htbhf.card.helper.TestConstants.HOMER_MOBILE;
-import static uk.gov.dhsc.htbhf.card.helper.TestConstants.SIMPSON_LAST_NAME;
+import static uk.gov.dhsc.htbhf.card.testhelper.AddressDTOTestDataFactory.aValidAddress;
+import static uk.gov.dhsc.htbhf.card.testhelper.TestConstants.HOMER_DOB;
```
Comment (gethyn1) [inline api/src/test/java/uk/gov/dhsc/htbhf/card/testhelper/CardDTOTestDataFactory.java]: Can `HOMER_DOB` be updated to `HOMER_DATE_OF_BIRTH` to match the other repos?

Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/card/testhelper/CardDTOTestDataFactory.java]: Done.

**Review by YetAnotherMatt — APPROVED** [review]

**Review by mikeSmith08 — APPROVED** [review]

**Pull request merged**